spirv_msl.cpp

```diff
@@ -13344,7 +13344,17 @@ void CompilerMSL::bitcast_from_builtin_load(uint32_t source_id, std::string &exp
 	}
 
 	if (expected_type != expr_type.basetype)
-		expr = bitcast_expression(expr_type, expected_type, expr);
+	{
+		if (expected_type == SPIRType::Half)
```

Comment:
The more appropriate thing to do here is to detect if there is a size mismatch and do value cast in that case. Should probably rename the function to reflect this.

Reply:
bitcast_to_builtin_store does an explicit check, can probably refactor that too while you're at it. Alternatively just do half -> float check.

Reply:
Refactored both of them.